SPEC/FILES.md

```diff
@@ -52,142 +57,342 @@ const files = [
+  - [Buffer][b], the raw Buffer of the cid
+  - String, the base58 encoded version of the cid
+- String, including the ipfs handler, a cid and a path to traverse to, ie:
+  - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66'
+  - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
+  - 'QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
+
+Returns a [Pull Stream][ps] with the contents of the file.
+
+```JavaScript
+const stream = ipfs.files.catPullStream(ipfsPath)
+// stream will be a stream containing the data of the file requested
 })
 ```
 
 A great source of [examples][] can be found in the tests for this API.
 
 #### `get`
 
-> Get [UnixFS][] files from IPFS.
+> Fetch a file or an entire directory tree from IPFS that is addressed by a valid IPFS Path.
 
 ##### `Go` **WIP**
 
 ##### `JavaScript` - ipfs.files.get(ipfsPath, [callback])
 
 ipfsPath can be of type:
 
-- `multihash` is a [multihash][] which can be passed as
-  - Buffer, the raw Buffer of the multihash
-  - String, the base58 encoded version of the multihash
-- String, including the ipfs handler, a multihash and a path to traverse to, ie:
+- [`cid`][cid] of type:
+  - [Buffer][b], the raw Buffer of the cid
+  - String, the base58 encoded version of the cid
+- String, including the ipfs handler, a cid and a path to traverse to, ie:
   - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66'
   - '/ipfs/QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
   - 'QmXEmhrMpbVvTh61FNAxP9nU7ygVtyvZA8HZDUaqQCAb66/a.txt'
 
-`callback` must follow `function (err, stream) {}` signature, where `err` is an
-error if the operation was not successful. `stream` will be a Readable stream in
-[*object mode*](https://nodejs.org/api/stream.html#stream_object_mode),
-outputting objects of the form
+`callback` must follow `function (err, files) {}` signature, where `err` is an error if the operation was not successful. `files` is an array containing objects of the following form:
 
 ```js
 {
   path: '/tmp/myfile.txt',
-  content: <Readable stream>
+  content: <data as a Buffer>
 }
 ```
 
-Here, each `path` corresponds to the name of a file, and `content` is a regular
-Readable stream with the raw contents of that file.
+Here, each `path` corresponds to the name of a file, and `content` is a regular Readable stream with the raw contents of that file.
 
-If no `callback` is passed, a promise is returned with the Readable stream.
+If no `callback` is passed, a promise is returned.
 
 **Example:**
 
 ```JavaScript
-const multihashStr = 'QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF'
+const validCID = 'QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF'
 
-ipfs.files.get(multihashStr, function (err, stream) {
-  stream.on('data', (file) => {
-    // write the file's path and contents to standard out
+ipfs.files.get(validCID, function (err, files) {
+  files.forEach((file) => {
     console.log(file.path)
-    file.content.pipe(process.stdout)
+    console.log(file.path.toString())
```

Comment:
Logging the path twice?
What about the path and filesize so we touched the content buffer as well.

Reply:
Thank you for catching that!